core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -20,13 +20,77 @@
   @version: 1.0
 **/
 
+const AWS = require('aws-sdk');
+const Uuid = require("uuid/v4");
+const request = require('request');
 const crypto = require('crypto');
 const logger = require("../components/logger.js");
-const config = require("../config/global_config.json")
+const config = require("../config/global_config.json");
 var truncate = require('unicode-byte-truncate');
 
 // Helper functions
 
+function assumeRole(configData, serviceData){
+  var isPrimary, roleArn, nonSlsApp = false;
+  if(serviceData){
+  	isPrimary = checkIsPrimary(serviceData.deployment_accounts[0].accountId, configData);
+  	roleArn = getRolePlatformService(serviceData.deployment_accounts[0].accountId, configData);
+  } 
+  var accessparams;
+  return new Promise((resolve, reject) => {
+    if (isPrimary || nonSlsApp) {
+        accessparams = {};
+        resolve(accessparams)
+      } else {
+        const sts = new AWS.STS({ region: process.env.REGION });
+        const roleSessionName = Uuid();
+        const params = {
+          RoleArn: roleArn,
+          RoleSessionName: roleSessionName,
+          DurationSeconds: 3600,
+        };
+        sts.assumeRole(params, (err, data) => {
+          if (err) {
+            reject({
+              "result": "serverError",
+              "message": "Unknown internal error occurred"
+            })
+          } else {
+            logger.info("Temporary Credentials are : " + JSON.stringify(data));
+            accessparams = {
+              accessKeyId: data.Credentials.AccessKeyId,
+              secretAccessKey: data.Credentials.SecretAccessKey,
+              sessionToken: data.Credentials.SessionToken,
+            };
+            resolve(accessparams)
+          }
+        })
+      }
+  })
+}
+
+function getLogsGroupsTags(logGroupName, tempCreds, serviceData) {
+  if(serviceData){
+  	tempCreds.region = serviceData.deployment_accounts[0].region;
+  }
+  var cloudwatchlogs = new AWS.CloudWatchLogs(tempCreds);
+  var params = {
+    logGroupName: logGroupName /* required */
+  };
+  return new Promise((resolve, reject) => {
+    cloudwatchlogs.listTagsLogGroup(params, function(err, data) {
+      if (err) {
+        logger.error("something went wrong while fetching tags..: " + JSON.stringify(err));
+        reject('error')
+      } else {
+        logger.debug('data ' + JSON.stringify(data))
```

Comment:
Need better logging.. ex: logger.debug(`tags for log group - ${logGroupName}: ${JSON.stringify(data)}`..

Reply:
Done.